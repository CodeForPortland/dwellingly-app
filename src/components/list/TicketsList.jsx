import React from 'react';
import PropTypes from 'prop-types';
import { Link, withRouter } from 'react-router-dom';
import Icon from '../icon/Icon';
import { formatDateFromString } from '../../utils';

function TicketsList({ items, match }) {
  return (
    <div className="list-group">
      {items &&
        items.length > 0 &&
        items.map(item => {
          const {
            id,
            issue,
            tenant,
            sender,
            sent,
            status,
            urgency,
            flagged
          } = item;
          const sentDate = formatDateFromString(sent);
          return (
            <Link
              key={id}
              to={`${match.path}/${id}`}
              className="list-group__item">
              <div className="contact-group">
                {status &&
                  urgency && (
                    <div
                      className={`contact-group__status contact-group__status--${urgency.toLowerCase()}`}
                    />
                  )}
                <h3 className="contact-group__name">{tenant.name}</h3>
                <p className="contact-group__title title">
                  {issue !== 'Compliment' &&
                    issue !== 'Resolved' && <Icon icon="comment" />}
                  {issue === 'Compliment' && <Icon icon="heart" />}
                  {issue}
                </p>
                <p>{tenant.address}</p>
                <p>Sender: {sender.name}</p>

                <div className="contact-group__meta ptr">
                  <time className="meta" dateTime={sentDate}>
                    {sentDate}
                  </time>
                  {status && <p className="status progress">{status}</p>}
                </div>

                {Boolean(flagged) && (
                  <div className="contact-group__flag pbr">
                    <Icon icon="flag" />
                  </div>
                )}
              </div>
            </Link>
          );
        })}
      {items.length === 0 && (
        <div className="list-group__item list-group__item--empty">
          <p>Empty state goes here...</p>
        </div>
      )}
    </div>
  );
}

TicketsList.propTypes = {
  match: PropTypes.shape({}),
  items: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      issue: PropTypes.string,
      tenant: PropTypes.shape({}),
      sender: PropTypes.shape({}),
      sent: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      status: PropTypes.string,
      urgency: PropTypes.string,
      flagged: PropTypes.string
    })
  )
};

TicketsList.defaultProps = {
  match: '',
  items: []
};

export default withRouter(TicketsList);
