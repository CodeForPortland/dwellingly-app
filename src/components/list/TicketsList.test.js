import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import TicketsList from './TicketsList';

describe('TicketsList Component', () => {
  it('renders empty state if items.length===0', () => {
    const TicketsListComponent = renderer.create(
      <MemoryRouter>
        <TicketsList />
      </MemoryRouter>
      ).toJSON();
    expect(TicketsListComponent).toMatchSnapshot();
  });
});
