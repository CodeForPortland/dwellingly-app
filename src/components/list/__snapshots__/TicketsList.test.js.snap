// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TicketsList Component renders empty state if items.length===0 1`] = `
<div
  className="list-group"
>
  <div
    className="list-group__item list-group__item--empty"
  >
    <p>
      Empty state goes here...
    </p>
  </div>
</div>
`;
